templates/activemq.service.erb

```diff
@@ -4,6 +4,9 @@ Description=Apache ActiveMQ
 [Service]
 Type=forking
 PIDFile=/opt/activemq/data/activemq.pid
+<%- if @jmx_enabled -%>
+Environment='ACTIVEMQ_SUNJMX_START=<%= @jmx_agent_opts %>'
+<%- end -%>
```

Comment:
do you tested it ?
With the syntax you use, you should have something like
```
Type=forking
PIDFile=/opt/activemq/data/activemq.pidEnvironment='ACTIVEMQ_SUNJMX_START=fooExecStart=/opt/activemq/bin/activemq start --extdir lib/tipaas
 ExecReload=/opt/activemq/bin/activemq restart --extdir lib/tipaas
```

Reply:
Yes, I tested it. if jmx_agent_opts is by default '', the file content is like below:
PIDFile=/opt/activemq/data/activemq.pid
Environment='ACTIVEMQ_SUNJMX_START='
ExecStart=/opt/activemq/bin/activemq start --extdir lib/tipaas


 if jmx_agent_opts is "-javaagent:/opt/jmx_exporter-%{hiera('monitoring::jmx_exporter::version')}/jmx_prometheus_javaagent-%{hiera('monitoring::jmx_exporter::version')}.jar=%{hiera('jmx_exporter_port')}:/opt/jmx_exporter-%{hiera('monitoring::jmx_exporter::version')}/activemq.yaml",
the content will be:

PIDFile=/opt/activemq/data/activemq.pid
Environment='ACTIVEMQ_SUNJMX_START=-javaagent:/opt/jmx_exporter-0.3.1/jmx_prometheus_javaagent-0.3.1.jar=9403:/opt/jmx_exporter-0.3.1/activemq.yaml'
ExecStart=/opt/activemq/bin/activemq start --extdir lib/tipaas

Reply:
Ok, thanks